albatross/core/model.h

```diff
@@ -63,35 +65,47 @@ using FoldIndexer = std::map<FoldName, FoldIndices>;
  * A combination of training and testing datasets, typically used in cross
  * validation.
  */
-template <class Feature>
+template <typename FeatureType>
 struct RegressionFold {
-  RegressionDataset<Feature> train;
-  RegressionDataset<Feature> test;
+  RegressionDataset<FeatureType> train_dataset;
+  RegressionDataset<FeatureType> test_dataset;
   FoldName name;
   FoldIndices test_indices;
 
-  RegressionFold(const RegressionDataset<Feature> &train_,
-                 const RegressionDataset<Feature> &test_,
+  RegressionFold(const RegressionDataset<FeatureType> &train_dataset_,
+                 const RegressionDataset<FeatureType> &test_dataset_,
                  const FoldName &name_, const FoldIndices &test_indices_)
-      : train(train_), test(test_), name(name_), test_indices(test_indices_){};
+      : train_dataset(train_dataset_), test_dataset(test_dataset_), name(name_), test_indices(test_indices_){};
 };
 
 /*
  * A model that uses a single Feature to estimate the value of a double typed
  * target.
  */
-template <class Feature>
+template <typename FeatureType>
 class RegressionModel : public ParameterHandlingMixin {
  public:
-  RegressionModel() : ParameterHandlingMixin(), has_been_fit_(false){};
+  typedef FeatureType Feature;
+  RegressionModel() : ParameterHandlingMixin(), has_been_fit_() {};
   virtual ~RegressionModel(){};
 
+  template <typename OtherFeatureType>
+  bool operator == (const RegressionModel<FeatureType> &other) const {
+    return false;
+  }
+
+  bool operator == (const RegressionModel<FeatureType> &other) const {
```

Comment:
are they equal if they `has_been_fit()` on different datasets?

Reply:
Yeah I guess right now they are ... which definitely isn't ideal.  I'll remove this class level operator and switch it to a unit test specific equality function.

Reply:
actually, I decided to change the equality to fail (hard) if the model has been fit and is being equality compared.  Idea being that if anyone is implementing a `RegressionModel` comparison will work fine as long as the model hasn't been fit (in which case parameter comparison should suffice) but if they want their model to be comparable after having been fit they'll need to write a custom equality operator which knows to compare any custom member variables which may have been modified during the fit.